### PR TITLE
Remove rest-hooks for specifications load

### PR DESCRIPTION
**Pull request by jamakase**
## What
Partially removes rest-hooks resources #10400

### CODE REVIEW

File: airbyte-webapp/package.json [hunk 1]
```diff
@@ -21,7 +21,6 @@
     "@fortawesome/free-solid-svg-icons": "^5.15.4",
     "@fortawesome/react-fontawesome": "^0.1.16",
     "@fullstory/browser": "^1.4.9",
-    "@rest-hooks/legacy": "^2.0.5",
```
Comment (timroes) [inline airbyte-webapp/package.json:24]: :tada:

**Comment by timroes** [comment]
Before we move this onto react-query I think we should solve the problem with relaoding data when focus changed, to prevent reloading the source definitions on focus changes.

Testing this also seems to break the app for me locally. Whenever you go to the "New Source" or "New destination" page, it will immediately try to load the source/destination specification, before the user selected an actual source/destination from the dropdown. It will then have `sourceDefinitionId` being empty and the query thus fail with a 400:


https://user-images.githubusercontent.com/877229/157407755-2e840e98-3618-4458-bd07-51f0c8c1d201.mp4

**Comment by Jamakase** [comment]
Fixed the problem with unnecessary call.

**Review by timroes — APPROVED** [review]
Code LGTM, tested locally to setup connection, seems not to break anything.

**Pull request merged**